src/app/store/general/types/base.ts

```diff
@@ -234,6 +234,13 @@ export type TLSCertificateState = {
   loading: boolean;
 };
 
+export type VaultEnabledState = {
+  errors: APIError;
+  data: boolean;
```

Comment:
Originally had this as a VaultEnabled type that I created, but this didn't return values properly and generally overcomplicated things. Although I'm happy to try and make that work if it's cleaner